### PR TITLE
Core: differentiate between unknown worlds and broken worlds in error message

### PR DESCRIPTION
**Pull request by Berserker66**
## What is this fixing or adding?
Makes it a bit clearer what broke, hopefully.

## How was this tested?
By adding carp at the top of factorio's init
[...]NameError: name 'carp' is not defined
[...]Exception: No functional world found to handle game Factorio. Did you mean 'factorio' (100% sure)? If so, it appears the world failed to initialize correctly.[...]

### CODE REVIEW

**Review by Ixrec — APPROVED** [review]
Added a typo in my fork to break my world's loading, and confirmed that attempting to generate from source fails with:

`Exception: No world found to handle game Outer Wilds. Did you mean 'Bumper Stickers' (40% sure)? Check your spelling or installation of that world.`

before this PR (admittedly, in the middle of a chain of exception stacks). After cherry-picking this PR's change onto my fork branch, that becomes:

`Exception: No functional world found to handle game Outer Wilds. Did you mean 'outer_wilds' (90% sure)? If so, it appears the world failed to initialize correctly.`

which is much clearer, thanks 👍

(and the code LGTM)

File: Generate.py
Comment (Jouramie) [inline Generate.py:450]: Feels like I'm nitpicking, but is it still worth mentioning the installation here, since the failed worlds are handled above?

**Review by Jouramie — APPROVED** [review]
I stumbled on this so many times in the early days of Stardew. That will be a great help to new contributors. 

LGTM

File: worlds/__init__.py
Comment (BadMagic100) [inline worlds/__init__.py:94]: I don't really like this because world names may not fuzzy match well against their file path e.g. HK, kh2.

But I don't know a better alternative so the code looks good enough to me 🤷

**Review by BadMagic100 — APPROVED** [review]

**Comment by BadMagic100** [comment]
Put into waiting on both author and core review since there is a question that may need addressing but was left with an approving review. Not sure if this is the best approach

File: worlds/__init__.py
Comment (black-sliver) [reply worlds/__init__.py:94]: Maybe we can change this once we add metadata to the worlds, but that's not a thing yet.

File: Generate.py
Comment (black-sliver) [reply Generate.py:450]: Idr exactly, but wasn't fuzzy matching case sensitive? If so, Factorio <-> factorio would be unable to match the 'F', which makes it less likely to find a good result.
Another idea to make this better would be to print "X worlds failed to load" in both code paths if  failed_world_loads isn't empty.

File: Generate.py
Comment (silasary) [inline Generate.py:448]: Would it be possible to add a line along the lines of "If you're filing a bug report, the actual error you're looking for is further up"?

Comment (Berserker66) [reply Generate.py:448]: I have another PR that stops execution at that first error.

**Review by black-sliver — APPROVED** [review]
Approving with the assuming we revisit this later once we have meta data. The biggest issue I see is that the world folder/apworld will always be lower case and the world name is likely upper case, so the error message may be misleading.

**Pull request merged**
